### PR TITLE
[pick_first] de-experiment pick first

### PR DESCRIPTION
**Pull request by mohanli-ml**
De-experiment pick first since we have both affinity and randomness E2E test running successfully.

### CODE REVIEW

**Comment by ctiller** [comment]
What impact will this have on prod?

**Comment by mohanli-ml** [comment]
> What impact will this have on prod?

I made this change for DirectPath cloud-to-prod. Currently when service owner specify `affinity` in their c2p_directpath config, gRPC will use ring hash instead of pick first as the LB policy. With this change, gRPC will use pick first as the default of affinity.

File: src/core/ext/filters/client_channel/lb_policy/pick_first/pick_first.cc [hunk 1]
```diff
@@ -62,10 +62,9 @@ namespace grpc_core {
 
 TraceFlag grpc_lb_pick_first_trace(false, "pick_first");
 
-// TODO(eostroukhov): Remove once this feature is no longer experimental.
 bool ShufflePickFirstEnabled() {
```
Comment (markdroth) [inline src/core/ext/filters/client_channel/lb_policy/pick_first/pick_first.cc:65]: Let's just remove this completely.  If it's passing interop tests, we no longer need the knob.

File: test/core/xds/xds_lb_policy_registry_test.cc [hunk 2]
```diff
@@ -486,13 +486,13 @@ TEST(PickFirst, ShuffleOmitted) {
   EXPECT_EQ(*result, "{\"pick_first\":{\"shuffleAddressList\":false}}");
 }
 
-TEST(PickFirst, EnvVarDisabled) {
+TEST(PickFirst, EnvVarEnabled) {
```
Comment (markdroth) [inline test/core/xds/xds_lb_policy_registry_test.cc:489]: This test no longer has value, so let's just remove it.

**Comment by markdroth** [comment]
> What impact will this have on prod?

As per the [gRFC](https://github.com/grpc/proposal/blob/master/A62-pick-first.md#temporary-environment-variable-protection), the env var guards two things:
1. Looking at the `shuffleAddressList` field in the pick_first LB policy config.
2. Accepting the pick_first LB policy config via xDS.

I don't think either one affects prod.

File: test/core/xds/xds_lb_policy_registry_test.cc (hunk 2 above)
Comment (mohanli-ml) [reply test/core/xds/xds_lb_policy_registry_test.cc]: Done!

File: src/core/ext/filters/client_channel/lb_policy/pick_first/pick_first.cc [hunk 3]
```diff
@@ -94,11 +83,7 @@ class PickFirstConfig : public LoadBalancingPolicy::Config {
   }
 
   void JsonPostLoad(const Json& /* json */, const JsonArgs& /* args */,
```
Comment (markdroth) [inline src/core/ext/filters/client_channel/lb_policy/pick_first/pick_first.cc:85]: This method can be removed.  It doesn't need to be provided if it's a no-op.

**Review by markdroth — COMMENTED** [review]
Please also remove all cases where we set the `GRPC_EXPERIMENTAL_PICKFIRST_LB_CONFIG` env var in tests.  Since the code no longer looks at that env var, there's no need for the tests to set it.

File: src/core/ext/filters/client_channel/lb_policy/pick_first/pick_first.cc (hunk 3 above)
Comment (mohanli-ml) [reply src/core/ext/filters/client_channel/lb_policy/pick_first/pick_first.cc]: Done. Please take another look. Thanks!

**Review by markdroth — APPROVED** [review]
Looks great!

**Pull request merged**